tools/ci/checks/get_applicable_changes.py

```diff
@@ -0,0 +1,99 @@
+#!/usr/bin/env python3
+
+import argparse
+import itertools
+import subprocess
+
+from typing import List
+
+
+def get_changed_files(git_reference: str) -> List[str]:
+    """
+    Get the list of files changed relative to the specified git reference.
+    """
+
+    cmd = ["git", "diff", git_reference, "--name-only"]
+
+    file_list = subprocess.check_output(cmd).decode("utf-8").split()
+
+    return file_list
+
+
+def ignore_extensions(input: List[str], extensions: str) -> List[str]:
+    if not extensions:
+        return input
+    extension_filter = tuple(f.strip() for f in extensions.split(","))
+
+    return list(filter(lambda f: not f.lower().endswith(extension_filter), input))
+
+
+def filter_extensions(input: List[str], extensions: str) -> List[str]:
+    if not extensions:
+        return input
+    extension_filter = tuple(f.strip() for f in extensions.split(","))
+
+    return list(filter(lambda f: f.lower().endswith(extension_filter), input))
+
+
+def filter_build_files_and_folders(input: List[str], limit_to_build: bool) -> List[str]:
+    """
+    Filter the list of files to those that would affect the build in some way.
+    """
+    if not limit_to_build:
+        return input
+    folders = ("cli/", "clivendor/", "govendor/", "sdk/", "testing/", "tools/")
+    files = ("conftest.py", "test_requirements.txt", "Dockerfile")
+
+    return list(filter(lambda f: f.startswith(folders + files), input))
+
+
+def flatten_file_list(file_args: List[str]) -> List[str]:
+
```

Comment:
What does pep8 say about this? I'd vote for no blank too.

Reply:
`pep8` (or `flake8`) doesn't complain as long as there are no internal double empty lines. My plan is to enable [black](https://github.com/ambv/black) at some point so that we don't need discussions about formatting anymore. It also only has 1 setting. Still want to be able to do this "automatically" though to reduce the developer burden.